Sources/SmokeDynamoDB/AWSDynamoDBClientConfiguration.swift

```diff
@@ -0,0 +1,195 @@
+                  awsRegion: awsRegion,
+                  endpointHostName: endpointHostNameOptional,
+                  endpointPort: endpointPort,
+                  requiresTLS: requiresTLS,
+                  service: service,
+                  contentType: contentType,
+                  target: target,
+                  traceContext: AWSClientInvocationTraceContext(),
+                  timeoutConfiguration: timeoutConfiguration,
+                  retryConfiguration: retryConfiguration,
+                  eventLoopProvider: eventLoopProvider,
+                  reportingConfiguration: reportingConfiguration,
+                  connectionPoolConfiguration: connectionPoolConfiguration)
+    }
+    
+    public func createOperationsClient(forTableName tableName: String)
+    -> AWSGenericDynamoDBTableOperationsClient<InvocationReportingType> {
+        return AWSGenericDynamoDBTableOperationsClient(config: self,
+                                                       tableName: tableName)
+    }
+    
+    internal func createClient() -> HTTPOperationsClient {
+        return HTTPOperationsClient(
+            endpointHostName: self.endpointHostName,
+            endpointPort: self.endpointPort,
+            contentType: self.contentType,
+            clientDelegate: self.clientDelegate,
+            timeoutConfiguration: self.timeoutConfiguration,
+            eventLoopProvider: .shared(self.eventLoopGroup),
+            connectionPoolConfiguration: self.connectionPoolConfiguration)
+    }
+    
+    internal func createAWSClient(logger: Logger, internalRequestId: String,
+                                  eventLoopOverride: EventLoop?, httpClientOverride: HTTPOperationsClient?)
+    -> _AWSDynamoDBClient<InvocationReportingType> {
+
+        let httpClient: HTTPOperationsClient
+        let ownsHttpClient: Bool
+        if let httpClientOverride = httpClientOverride {
+            httpClient = httpClientOverride
+            ownsHttpClient = false
+        } else {
+            httpClient = createClient()
+            ownsHttpClient = true
+        }
+        
+        return _AWSDynamoDBClient(credentialsProvider: self.credentialsProvider,
+                                  awsRegion: self.awsRegion,
+                                  reporting: self.reportingProvider(logger, internalRequestId, eventLoopOverride),
+                                  httpClient: httpClient,
+                                  ownsHttpClient: ownsHttpClient,
+                                  retryConfiguration: self.retryConfiguration,
+                                  reportingConfiguration: self.reportingConfiguration,
+                                  retryOnErrorProvider: { error in error.isRetriable() })
+    }
+    
+    internal func createAWSClient<OverrideInvocationReportingType: HTTPClientCoreInvocationReporting>(
+        reporting: OverrideInvocationReportingType,
+        httpClient: HTTPOperationsClient?)
```

Comment:
Maybe rename to `httpClientOverride` to match the previous function.

Reply:
Fixed